### PR TITLE
Start up sequence callback

### PR DESCRIPTION
**Pull request by jigar-f**

### CODE REVIEW

**Comment by jigar-f** [comment]
@atavism Go ahead and review this as well, If all is okay merge this and then put out new build.

**Comment by atavism** [comment]
Taking a look, @jigar-f!

File: android/app/src/main/java/org/getlantern/mobilesdk/util/DnsDetector.java [hunk 1]
```diff
@@ -61,14 +61,12 @@ public DnsDetector(Context context, String fakeDnsIP) {
                     public void onAvailable(@NonNull Network network) {
                         Logger.debug(TAG, "Adding available network");
                         allNetworks.put(network, "");
-                        EventBus.getDefault().postSticky(Event.NetworkAvailable);
```
Comment (atavism) [inline android/app/src/main/java/org/getlantern/mobilesdk/util/DnsDetector.java:64]: Not necessary on this PR, but we should just remove the EventBus stuff entirely

Comment (jigar-f) [reply android/app/src/main/java/org/getlantern/mobilesdk/util/DnsDetector.java:64]: Yep, eventually we will migrate to use gomodel so most of this kotlin code will be removed.

File: go.mod [hunk 2]
```diff
@@ -2,7 +2,7 @@ module github.com/getlantern/lantern-client
 
 go 1.22.0
 
-// replace github.com/getlantern/flashlight/v7 => ../flashlight
+replace github.com/getlantern/flashlight/v7 => ../flashlight
```
Comment (atavism) [inline go.mod:5]: I think you want to comment this out, right?

**Comment by atavism** [comment]
Just checking, the VPN switch looks different to me with these changes. Is that intentional?

Before:

image

After:
Screenshot 2024-05-27 at 4 57 21 PM

File: lib/common/ui/custom/dialog.dart [hunk 3]
```diff
@@ -47,6 +47,110 @@ class CDialog extends StatefulWidget {
     ).show(context);
   }
 
+  static void showInternetUnavailableDialog(BuildContext context) {
```
Comment (atavism) [inline lib/common/ui/custom/dialog.dart:50]: I confirmed I see this when I turn on Airplane mode:

Screenshot_20240527-172916

File: lib/app.dart [hunk 4]
```diff
@@ -113,80 +121,84 @@ class _LanternAppState extends State<LanternApp> {
...
+                        Locale('fr', 'FR'),
+                        Locale('en', 'US'),
+                        Locale('fa', 'IR'),
+                        Locale('th', 'TH'),
+                        Locale('ms', 'MY'),
+                        Locale('ru', 'RU'),
+                        Locale('ur', 'IN'),
+                        Locale('zh', 'CN'),
+                        Locale('zh', 'HK'),
+                        Locale('es', 'ES'),
```
Comment (atavism) [inline lib/app.dart:190]: I think we want to keep `Locale('es', 'CU')`

File: lib/vpn/vpn_tab.dart [hunk 5]
```diff
@@ -1,17 +1,22 @@
...
 import 'vpn_status.dart';
 import 'vpn_switch.dart';
 
 class VPNTab extends StatelessWidget {
-  VPNTab({Key? key}) : super(key: key);
+  const VPNTab({Key? key}) : super(key: key);
 
   @override
   Widget build(BuildContext context) {
+    final vpnModel = context.watch<VPNChangeNotifier>();
```
Comment (atavism) [inline lib/vpn/vpn_tab.dart:19]: Maybe rename this to vpnNotifier? I initially thought we were calling isFlashlightInitialized on this [vpnModel](https://github.com/getlantern/lantern-client/blob/main/lib/vpn/vpn_model.dart)

**Review by atavism — APPROVED** [review]
Just a few comments but otherwise LGTM

File: go.mod (hunk 2 above)
Comment (jigar-f) [reply go.mod:5]: Yep, Forgot to uncomment it. Done.

**Comment by jigar-f** [comment]
> This is how the VPN switch looks for me on desktop with these changes. Just confirming we want to change the appearance like this?

I had to change the switch package due to lack of customization else the UI same.
cc @Derekf5 Do you mind checking this UI and letting me know, if these changes look good.

File: lib/common/ui/custom/dialog.dart (hunk 3 above)
Comment (jigar-f) [reply lib/common/ui/custom/dialog.dart:50]: @atavism we are showing this dialog when there is no interconnection found and VPN switch disable.

File: lib/vpn/vpn_tab.dart (hunk 5 above)
Comment (jigar-f) [reply lib/vpn/vpn_tab.dart:19]: Changing....

File: lib/app.dart (hunk 4 above)
Comment (jigar-f) [reply lib/app.dart:190]: Added

**Comment by atavism** [comment]
@Derekf5 @jigar-f Are we fine changing the appearance of the switch in the desktop app to look like this? (IMO, it's a little too big). I think that's the main holdup now getting this one merged.

image

**Comment by jigar-f** [comment]
I think Derekf approved these changes last week.

**Comment by jigar-f** [comment]
Also @atavism We need to merge [flashlight PR](https://github.com/getlantern/flashlight/pull/1372) before this

**Comment by atavism** [comment]
> I think Derekf approved these changes last week.

Cool, thanks for confirming!

**Comment by atavism** [comment]
@jigar-f https://github.com/getlantern/flashlight/pull/1372 has been merged and this PR has been updated with the latest changes. OK to merge this one now?

**Comment by atavism** [comment]
Done!

**Pull request merged**